flask_resty/view.py

```diff
@@ -415,20 +416,45 @@ def commit(self):
     def resolve_integrity_error(self, error):
         original_error = error.orig
 
-        if (
-            hasattr(original_error, 'pgcode') and
-            original_error.pgcode in (
+        if hasattr(original_error, 'pgcode'):
+            if original_error.pgcode in (
                 '23502',  # not_null_violation
                 '23514',  # check_violation
-            )
-        ):
-            # Using the psycopg2 error code, we can tell that this was not from
-            # an integrity error that was not a conflict. This means there was
-            # a schema bug, so we emit an interal server error instead.
-            return error
+            ):
+                # Using the psycopg2 error code, we can tell that this was not
+                # from an integrity error that was not a conflict. This means
+                # there was a schema bug, so we emit an internal server error
+                # instead.
+                return error
+            else:
+                diag = original_error.diag
+
+                schema_name = diag.schema_name
+                table_name = diag.table_name
+                constraint_name = diag.constraint_name
+
+                insp = sa.inspect(self.session.bind)
```

Comment:
does it make more sense to do this at startup for each table, or to cache these?